### ISSUE TITLE
fine-tune栗子

### ISSUE THREAD

**Issue by congmonkey** [bob48523/modified_wrn#2]
https://github.com/Spandan-Madan/Pytorch_fine_tuning_Tutorial

**Comment by congmonkey** [comment]
class newmodel(nn.Module):
            def __init__(self):
                super(newmodel, self).__init__()
                self.features = nn.Sequential(
                    # stop at conv4
                    *list(net.features.children())[:-3]
                )
                
                
            def forward(self, x):
                x = self.features(x)
                #x = F.dropout(x, p=self.dropout, inplace=False, training=self.training) 
                return x
我要修改网络结构, 比如在原来fc层和avgpool前加东西, *list(net.features.children())[:-3] 这里要写多少(-2?)

**Comment by congmonkey** [comment]
网络长这样
class Desnetx_se(nn.Module):
    def __init__(self, block, num_blocks, cardinality, reduction=0.5, num_classes=10):
        super(Desnetx_se, self).__init__()
        self.cardinality = cardinality
        self.in_planes = 64
        
        width = [64, 128, 256]
        
        self.conv1 = nn.Conv2d(3, self.in_planes, kernel_size=3, padding=1, bias=False)
        self.bn1 = nn.BatchNorm2d(self.in_planes)
        
         
        self.layer1 = self._make_layer(block, self.in_planes, width[0], num_blocks[0], stride=1)
        out_planes = int(math.floor((self.in_planes+self.last_in_planes)*reduction))
        self.trans1 = Transition(self.in_planes+self.last_in_planes, out_planes)
        self.se1 = SE(out_planes)
        
        
        self.layer2 = self._make_layer(block, out_planes, width[1], num_blocks[1], stride=1)
        out_planes = int(math.floor((self.in_planes+self.last_in_planes)*reduction))
        self.trans2 = Transition((self.in_planes+self.last_in_planes), out_planes)
        self.se2 = SE(out_planes)
        
        self.layer3 = self._make_layer(block, out_planes, width[2], num_blocks[2], stride=1)      
        self.se3 = SE((self.in_planes+self.last_in_planes))       
        
        
       # self.layer4 = self._make_layer(block, width[3], num_blocks[3], stride=2)
    
        self.fc = nn.Linear((self.in_planes+self.last_in_planes), num_classes)
        
        for m in self.modules():
            if isinstance(m, nn.Conv2d):
                n = m.kernel_size[0] * m.kernel_size[1] * m.out_channels
                m.weight.data.normal_(0, math.sqrt(2. / n))
            elif isinstance(m, nn.BatchNorm2d):
                m.weight.data.fill_(1)
                m.bias.data.zero_()
            elif isinstance(m, nn.Linear):
                m.bias.data.zero_()

    def _make_layer(self, block, in_planes, planes, num_blocks, stride):
        strides = [stride] + [1]*(num_blocks-1)
        layers = []
        self.last_in_planes = in_planes
        self.in_planes = in_planes
        for stride in strides:
            layers.append(block(self.in_planes, planes, self.cardinality, stride))
            self.in_planes = planes * block.widen_factor
        #self.in_planes += self.last_in_planes 
        return nn.Sequential(*layers)


    def forward(self, x):
        global FLOPS
        FLOPS = 0
        out = F.relu(self.bn1(self.conv1(x)))
        FLOPS += self.conv1.in_channels*self.conv1.out_channels*self.conv1.kernel_size[0]*self.conv1.kernel_size[1]*out.size(2)*out.size(3)
        
        out1 = self.layer1(out)
        out = torch.cat((out1, out),1) 
        out = self.trans1(out)
        out = self.se1(out)
        
        out1 = self.layer2(out)
        out = torch.cat((out1, out),1)      
        out = self.trans2(out)
        out = self.se2(out)

        out1 = self.layer3(out)
        out = torch.cat((out1, out),1)
        out = self.se3(out)

       # out = self.layer4(out)              
            
            
        out = F.avg_pool2d(out, 8)
        out = out.view(out.size(0), -1)
        

        
        out = self.fc(out)
        FLOPS += self.fc.in_features*self.fc.out_features 
        
        return out
是要算init里面定义的结构 还是forward??
